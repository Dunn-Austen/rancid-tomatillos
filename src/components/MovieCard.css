@import url(https://fonts.googleapis.com/css?family=Exo+2:200i);
@import url(https://fonts.googleapis.com/css?family=Hanalei+Fill&display=swap);

.welcome-user {
  /* animation: flicker 3s infinite alternate; */
  border-radius: 20px;
  color: #C654DE;
  font-family: 'Exo 2', sans-serif;
  font-size: 25px;
}

.poster-title  {
  background-color: #2D2D2F;
  border: 2px ridge #fff;
  border-radius: 20px;
  box-shadow: 0 0 15px 2px black;
  color: #c654dd;
  font-family: 'Exo 2', sans-serif;
  font-size: 25px;
  text-align: center;
  text-transform: uppercase;
  width: 330px;
}

/* Replace root color styling variables with hexidecimal*/
:root {
  --neon-text-color: purple;
  --neon-border-color: #2D2D2F;
}

@keyframes flicker {
  0%, 19%, 21%, 23%, 25%, 54%, 56%, 100% {
    text-shadow:
        -0.2rem -0.2rem 1rem #fff,
        0.2rem 0.2rem 1rem #fff,
        0 0 2rem var(--neon-text-color),
        0 0 4rem var(--neon-text-color),
        0 0 6rem var(--neon-text-color),
        0 0 8rem var(--neon-text-color),
        0 0 10rem var(--neon-text-color);

    box-shadow:
        0 0 .5rem #fff,
        inset 0 0 .5rem #fff,
        0 0 2rem var(--neon-border-color),
        inset 0 0 2rem var(--neon-border-color),
        0 0 4rem var(--neon-border-color),
        inset 0 0 4rem var(--neon-border-color);
  }

  20%, 24%, 55% {
      text-shadow: none;
      box-shadow: none;
  }
}

.movie-card {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: 10px 0;
  width: 350px
}

.poster {
  box-shadow: 0px 0px 20px 5px black;
  height: 450px;
  width: 350px;
}

.avg-container {
  text-shadow: 3px 3px 3px black;
  width: 170px;
}

.avg-container:before {
  color: #c654de;
  content: 'Average ';
  font-family: 'Hanalei Fill', cursive;
  position: relative;
  bottom: 20px;
  font-size: 14px;
}

.avg-container:after {
  color: #c654de;
  content: 'Rating';
  font-family: 'Hanalei Fill', cursive;
  position: relative;
  bottom: 20px;
  font-size: 14px;
}

.average-img {
  border-radius: 85px;
  height: 70px;
  margin-top: -30px;
  text-shadow: 3px 3px 3px black;
}

.average-score {
  color: #3B397F;
  font-size: 20px;
  position: relative;
  text-align: center;
  text-shadow: 0px 0px 1px black;
  top: -42px;
  left: 4px;
  z-index: 1;
}

.view-movie {
  color: #d34ee5;
  font-size: 25px;
  margin-top: -23px;
  text-align: center;
  text-shadow: 0px 0px 14px black;
  font-weight: bold;
  text-decoration: underline;
  text-decoration-style: solid;
  text-decoration-color: silver;
  width: 140px;
}

.view-movie:hover {
  color: #ED7899;
  text-decoration: none;
  text-shadow: 6px 6px 3px black;
}
