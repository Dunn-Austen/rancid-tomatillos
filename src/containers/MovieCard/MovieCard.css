@import url(https://fonts.googleapis.com/css?family=Exo+2:200i);
@import url(https://fonts.googleapis.com/css?family=Hanalei+Fill&display=swap);

.poster-title  {
  background-color: #2D2D2F;
  border: 2px ridge #fff;
  border-radius: 20px;
  box-shadow: 0 0 15px 2px black;
  color: #c654dd;
  font-family: 'Exo 2', sans-serif;
  font-size: 25px;
  text-align: center;
  text-transform: uppercase;
  width: 330px;
}

.poster-title:hover {
  box-shadow: 0 0 6px white;
  content: 'View Movie';
}

.movie-card {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: 10px 0;
  width: 350px
}

.poster {
  box-shadow: 0px 0px 20px 5px black;
  height: 450px;
  width: 350px;
}

.avg-container {
  text-shadow: 3px 3px 3px black;
  width: 200px;
}

.avg-container:before {
  bottom: 20px;
  color: #c654de;
  content: 'Average ';
  font-family: 'Hanalei Fill', cursive;
  font-size: 17px;
  float: left;
  margin-right: 4px;
}

.avg-container:after {
  bottom: 20px;
  color: #c654de;
  content: 'Rating';
  font-family: 'Hanalei Fill', cursive;
  font-size: 17px;
  float: right;
  margin-right: 4px;
}

.average-img {
  border-radius: 85px;
  height: 70px;
  margin-top: -30px;
  margin-left: 4px;
  text-shadow: 3px 3px 3px black;
}

.average-score {
  color: #3B397F;
  font-size: 20px;
  font-weight: bold;
  left: 4px;
  position: relative;
  text-align: center;
  text-shadow: 0px 0px 1px purple;
  top: -42px;
  z-index: 1;
}

.view-movie {
  color: #ED7899;
  font-size: 26px;
  font-weight: bold;
  margin-top: -23px;
  text-align: center;
  text-shadow: 6px 6px 3px black;
  width: 140px;
}

.view-movie:hover {
  color: #D45DC1;
  text-shadow: 10px 15px 3px black;
}

.star-container {
  height: 85px;
  margin-top: -85px;
  position: relative;
  text-shadow: 3px 3px 3px black;
  top: 65px;
  width: 200px;
  z-index: 1;
}

.star-container:before {
  bottom: 20px;
  color: white;
  content: 'Your ';
  font-family: 'Hanalei Fill', cursive;
  font-size: 18px;
  left: 29px;
  position: relative;
}

.star-container:after {
  bottom: 20px;
  color: white;
  content: 'Score';
  font-family: 'Hanalei Fill', cursive;
  font-size: 17px;
  left: -10px;
  position: relative;
}

.user-star {
  height: 80px;
  position: relative;
  left: 21px;
}

.user-score {
  color: #3B397F;
  font-size: 20px;
  font-weight: bold;
  left: -32px;
  position: relative;
  text-align: center;
  text-shadow: 0px 0px 1px pink;
  top: -26px;
  z-index: 2;
}

.user__score--two {
  color: #3B397F;
  font-size: 20px;
  font-weight: bold;
  left: -26px;
  margin-right: 9px;
  position: relative;
  text-align: center;
  text-shadow: 0px 0px 1px pink;
  top: -26px;
  z-index: 2;
}
