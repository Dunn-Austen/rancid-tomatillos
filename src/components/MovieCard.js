import React from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import './MovieCard.css';

const MovieCard = props => {
  const {
    id,
    title,
    poster_path,
    backdrop_path,
    release_date,
    overview,
    average_rating
  } = props;
  return (
    <div>
      <h1>{title}</h1>
      <img
        className='poster'
        src={poster_path}
        alt={`Poster Picture of ${title}`}
      />
      <NavLink to={`/movies/${id}`}>
        <div>View Movie</div>
      </NavLink>
    </div>
  );
};

const mapStateToProps = state => ({
  allMovies: state.movies
});

export default connect(mapStateToProps, null)(MovieCard);
