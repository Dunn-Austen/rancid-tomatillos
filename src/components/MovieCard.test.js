import React from 'react';
import { shallow } from 'enzyme';
import { MovieCard } from './MovieCard';

describe('MovieCard', () => {
  let wrapper;
  it('should match the MovieCard Snapshot', () => {
    wrapper = shallow(
      <MovieCard
        id={3}
        title={'Frozen II'}
        poster_path={
          'https://image.tmdb.org/t/p/original//pjeMs3yqRmFL3giJy4PMXWZTTPa.jpg'
        }
        backdrop_path={
          'https://image.tmdb.org/t/p/original//xJWPZIYOEFIjZpBL7SVBGnzRYXp.jpg'
        }
        release_date={'2019-11-20'}
        overview={
          'Elsa, Anna, Kristoff and Olaf head far into the forest to learn the truth about an ancient mystery of their kingdom.'
        }
        average_rating={8}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
