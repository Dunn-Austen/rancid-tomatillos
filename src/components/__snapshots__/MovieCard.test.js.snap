// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should match the MovieCard Snapshot 1`] = `
<div
  className="movie-card"
>
  <h1
    className="poster-title"
  >
    Frozen II
  </h1>
  <img
    alt="Poster Picture of Frozen II"
    className="poster"
    src="https://image.tmdb.org/t/p/original//pjeMs3yqRmFL3giJy4PMXWZTTPa.jpg"
  />
  <p
    className="average-rating"
  >
    Average Rating: 
    8
  </p>
  <NavLink
    to="/movies/3"
  >
    <div
      className="view-name"
    >
      View Movie
    </div>
  </NavLink>
</div>
`;
